pkg/server/claims.go

```diff
@@ -139,22 +145,27 @@ func parsePrivateClaims(oidcToken jwt.Token) (*oidcClaims, error) {
 	}
 	claims.Repository = r
 
-	claims.Ref = optionalClaim(oidcToken, "ref")
-	claims.RefType = optionalClaim(oidcToken, "ref_type")
-	claims.Sha = optionalClaim(oidcToken, "sha")
-	claims.RepositoryID = optionalClaim(oidcToken, "repository_id")
-	claims.RepositoryOwner = optionalClaim(oidcToken, "repository_owner")
-	claims.RepositoryOwnerID = optionalClaim(oidcToken, "repository_owner_id")
-	claims.RunID = optionalClaim(oidcToken, "run_id")
-	claims.RunNumber = optionalClaim(oidcToken, "run_number")
-	claims.Actor = optionalClaim(oidcToken, "actor")
-	claims.ActorID = optionalClaim(oidcToken, "actor_id")
-	claims.EventName = optionalClaim(oidcToken, "event_name")
-	claims.Workflow = optionalClaim(oidcToken, "workflow")
-	claims.WorkflowRef = optionalClaim(oidcToken, "workflow_ref")
-	claims.WorkflowSha = optionalClaim(oidcToken, "workflow_sha")
-	claims.JobWorkflowRef = optionalClaim(oidcToken, "job_workflow_ref")
-	claims.JobWorkflowSha = optionalClaim(oidcToken, "job_workflow_sha")
+	claims.Ref = optionalClaim[string](oidcToken, "ref")
+	claims.RefType = optionalClaim[string](oidcToken, "ref_type")
+	claims.Sha = optionalClaim[string](oidcToken, "sha")
+	claims.RepositoryID = optionalClaim[string](oidcToken, "repository_id")
+	claims.RepositoryOwner = optionalClaim[string](oidcToken, "repository_owner")
+	claims.RepositoryOwnerID = optionalClaim[string](oidcToken, "repository_owner_id")
+	claims.RunID = optionalClaim[string](oidcToken, "run_id")
+	claims.RunNumber = optionalClaim[string](oidcToken, "run_number")
+	claims.Actor = optionalClaim[string](oidcToken, "actor")
+	claims.ActorID = optionalClaim[string](oidcToken, "actor_id")
+	claims.EventName = optionalClaim[string](oidcToken, "event_name")
+	claims.Workflow = optionalClaim[string](oidcToken, "workflow")
+	claims.WorkflowRef = optionalClaim[string](oidcToken, "workflow_ref")
+	claims.WorkflowSha = optionalClaim[string](oidcToken, "workflow_sha")
+	claims.JobWorkflowRef = optionalClaim[string](oidcToken, "job_workflow_ref")
+	claims.JobWorkflowSha = optionalClaim[string](oidcToken, "job_workflow_sha")
+
+	emailVerified := optionalClaim[bool](oidcToken, "email_verified")
+	if emailVerified {
+		claims.Email = optionalClaim[string](oidcToken, "email")
```

Comment:
We should probably have a test for this special behavior

Reply:
Agree

Reply:
Added test for this and also just for this function in general.

Reply:
Is there a test where email_verified is false? I'm not seeing it